### DISCUSSION TITLE
Langchain fast-api streaming to next js client

### DISCUSSION THREAD

**Discussion by jasan-s** [vercel/community#2962] [Help] [answered]
I am hoping to deploy langchain built chatbot to vercel with fast-api and connect to my client app to stream responses from LLM. 

How would this be feasible.

**Comment by amyegan** [comment, marked as answer]
Hi @jasan-s! We actually have a few templates that seem like good starters for what you're trying to build:
- [AgentGPT](https://vercel.com/templates/next.js/agent-gpt)
- [Next.js AI Chatbot](https://vercel.com/templates/next.js/nextjs-ai-chatbot)
- [Next.js FastAPI Starter](https://vercel.com/templates/next.js/nextjs-fastapi-starter)

Even if you don't directly use one of the templates, they are useful as a proof-of-concept. You can find more template/example options at https://vercel.com/templates

**Reply by jasan-s** [reply]
@amyegan  thanks, actually specifically i wanted to learn if Vercel supports python runtime response streaming to client? I have tried langchain JS within NExt JS api function but streaming did not work when deployed to vercel. I had to modify to run on edge function. 

I wanted to switch to using Langchain python since it is more featured and concerned if vercel supports python streaming response during runtime in their serveless functions.

**Reply by amyegan** [reply]
The Python runtime does not currently supporting streaming responses on Vercel, but this feature request was shared with the team 🙂
